### ISSUE TITLE
FTBFS against libpng 1.6

### ISSUE THREAD

**Issue by sunweaver** [ArcticaProject/nx-libs#100]
@mbecroft just encoutered the below FTBFS...

```
12:42 < mbecroft> Pgn.cpp:417:15: error: invalid use of incomplete type 'png_info {aka struct png_info_def}'
12:43 < mbecroft>    if (infoPtr -> color_type == PNG_COLOR_TYPE_PALETTE)
```

Presumably some libpng 1.5 -> 1.6 issue (as I build against libpng 1.5.7 whereas mbecroft builds against 1.6 on Gentoo).

**Comment by sunweaver** [comment]
From http://www.libpng.org/pub/png/libpng.html (20160302):

```
The libpng 1.5.x and 1.6.x series continue the evolution of the libpng API, finally hiding the contents of
the venerable and hoary png_struct and png_info data structures inside private (i.e., non-installed) 
header files. Instead of direct struct-access, applications should be using the various png_get_xxx() 
and png_set_xxx() accessor functions, which have existed for almost as long as libpng itself. 
```

**Comment by mbecroft** [comment]
Already fixed in 3.6.x branch. Closing. Resolved by 79f218b563f6a4d3b17e92ee85ad242575daf8da
